### PR TITLE
[Task 1872400] Library for azure network service

### PR DESCRIPTION
**Pull request by Flanker32**
### Content
This PR add library for Azure network service, based on framework in https://github.com/microsoft/azure-maven-plugins/pull/1740, prepare for virtual machine support in azure tooling.

### CODE REVIEW

File: azure-toolkit-libs/azure-toolkit-compute-lib/src/main/java/com/microsoft/azure/toolkit/lib/compute/ip/AzurePublicIpAddress.java [hunk 1]
```diff
@@ -0,0 +1,51 @@
...
+    public PublicIpAddress get(@Nonnull final String subscriptionId, @Nonnull final String resourceGroup, @Nonnull final String name) {
+        final PublicIpAddresses virtualMachinesManager = getPublicIpAddressManager(subscriptionId);
+        return new PublicIpAddress(virtualMachinesManager.getByResourceGroup(resourceGroup, name), this);
+    }
+
+    public PublicIpAddress create(@Nonnull final DraftPublicIpAddress draftPublicIpAddress) {
+        return draftPublicIpAddress.create(this);
+    }
+
+    public PublicIpAddresses getPublicIpAddressManager(String subscriptionId) {
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-compute-lib/src/main/java/com/microsoft/azure/toolkit/lib/compute/ip/AzurePublicIpAddress.java:43]: remove public

File: azure-toolkit-libs/azure-toolkit-compute-lib/src/main/java/com/microsoft/azure/toolkit/lib/compute/ip/DraftPublicIpAddress.java [hunk 2]
```diff
@@ -0,0 +1,54 @@
...
+                .withExistingResourceGroup(resourceGroup)
+                .withLeafDomainLabel(leafDomainLabel)
+                .create();
+        refreshStatus();
+        return this;
+    }
+
+    @Override
+    protected String loadStatus() {
+        return Optional.ofNullable(module).map(ignore -> super.loadStatus()).orElse(Status.DRAFT);
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-compute-lib/src/main/java/com/microsoft/azure/toolkit/lib/compute/ip/DraftPublicIpAddress.java:42]: use remote

File: azure-toolkit-libs/azure-toolkit-compute-lib/src/main/java/com/microsoft/azure/toolkit/lib/compute/network/DraftNetwork.java [hunk 3]
```diff
@@ -0,0 +1,74 @@
...
+    private String resourceGroup;
+    private String name;
+    private Region region;
+    private String addressSpace;
+
+    private String subnet;
+    private String subnetAddressSpace;
+
+    public DraftNetwork(@Nonnull final String subscriptionId, @Nonnull final String resourceGroup, @Nonnull final String name) {
+        super(getResourceId(subscriptionId, resourceGroup, name), null);
```
Comment (andxu) [inline azure-toolkit-libs/azure-toolkit-compute-lib/src/main/java/com/microsoft/azure/toolkit/lib/compute/network/DraftNetwork.java:38]: no id to create id for draft

Comment (Flanker32) [reply azure-toolkit-libs/azure-toolkit-compute-lib/src/main/java/com/microsoft/azure/toolkit/lib/compute/network/DraftNetwork.java:35]: Let's redesign the draft pattern and apply to all resources later

**Review by andxu — APPROVED** [review]
LGTM

**Pull request merged**